### PR TITLE
add bidirectional mode

### PR DESCRIPTION
**Pull request by SigridHo**

### CODE REVIEW

**Review by haomingchan0811 — COMMENTED** [review]
Minor style modification

File: textsum/main.py [hunk 1]
```diff
@@ -260,11 +262,12 @@ def vec2text_from_full():
     argparser.add_argument('--learning_rate', type=float, default=0.003)
     argparser.add_argument('--teach_ratio', type=float, default=1)
     argparser.add_argument('--dropout', type=float, default=0.0)
-    argparser.add_argument('--attn_model', type=str, choices=['none', 'general', 'dot'], default='none')
+    argparser.add_argument('--attn_model', type=str, choices=['none', 'general', 'dot', 'concat'], default='none')
     argparser.add_argument('--show_attn', action='store_true', default = False)
     # argparser.add_argument('--max_norm', type=float, default=100.0)
     argparser.add_argument('--l2', type=float, default=0.03)
     argparser.add_argument('--rnn_model', type=str, choices=['gru', 'lstm'], default='gru')
+    argparser.add_argument('--bidir', action='store_true', default = False)
```
Comment (boche) [inline textsum/main.py]: use_bidir

File: textsum/model/DecoderRNN.py [hunk 2]
```diff
@@ -18,6 +18,7 @@ def __init__(self, vocab_size, emb, hidden_size, nlayers, teach_ratio,
         self.teach_ratio = teach_ratio
         self.attn_model = attn_model
         self.rnn_model = rnn_model
+        self.bidir = bidir
```
Comment (boche) [inline textsum/model/DecoderRNN.py]: use_bidir

File: textsum/model/DecoderRNN.py [hunk 3]
```diff
@@ -32,9 +33,10 @@ def __init__(self, vocab_size, emb, hidden_size, nlayers, teach_ratio,
             self.rnn = nn.LSTM(input_size = rnn_input_size, hidden_size = hidden_size,
                     dropout = dropout, num_layers = nlayers, batch_first = True)
         if self.attn_model != 'none':
-            self.concat = nn.Linear(hidden_size * 2, hidden_size)
+            self.concat_size = hidden_size * (3 if bidir else 2)
```
Comment (boche) [inline textsum/model/DecoderRNN.py]: concat_size, no need to use self.

File: textsum/model/attn.py [hunk 4]
```diff
@@ -5,24 +5,39 @@
...
-    def __init__(self, method, hidden_size):
+    def __init__(self, method, hidden_size, bidir):
         super(Attn, self).__init__()
         self.method = method
         self.hidden_size = hidden_size
+        self.bidir = bidir
+        
         if self.method == 'general':
-            self.attn = nn.Linear(self.hidden_size, hidden_size)
+            self.encoder_output_size = self.hidden_size * (2 if bidir else 1)
```
Comment (boche) [inline textsum/model/attn.py]: no self. for encoder_output_size

**Pull request merged**